src/MediatR/INotificationHandler.cs

```diff
@@ -27,32 +27,14 @@ public abstract class NotificationHandler<TNotification> : INotificationHandler<
     {
         public Task Handle(TNotification notification, CancellationToken cancellationToken)
         {
-            Handle(notification);
+            HandleCore(notification);
             return Unit.Task;
         }
 
         /// <summary>
         /// Override in a derived class for the handler logic
         /// </summary>
         /// <param name="notification">Notification</param>
-        protected abstract void Handle(TNotification notification);
-    }
-
-    /// <summary>
-    /// Wrapper class for an async notification handler, ignoring the cancellation token
-    /// </summary>
-    /// <typeparam name="TNotification">The notification type</typeparam>
-    public abstract class AsyncNotificationHandler<TNotification> : INotificationHandler<TNotification>
-        where TNotification : INotification
-    {
-        public Task Handle(TNotification notification, CancellationToken cancellationToken)
-            => Handle(notification);
-
-        /// <summary>
-        /// Override in a derived class for the handler logic 
-        /// </summary>
-        /// <param name="notification">Notification</param>
-        /// <returns>A task</returns>
-        protected abstract Task Handle(TNotification notification);
+        protected abstract void HandleCore(TNotification notification);
```

Comment:
Why did these become `HandleCore` again?

Reply:
Also, the async notification handler should probably also forward the cancellation token?

Reply:
So the `AsyncRequestHandler` class now has an overload that differs only in return type. I could:

 - Make all the Handle methods be explicit interface implementations, so I can keep the overload
 - Have only this class have `HandleCore`
 - Get rid of all these helper classes because does it really matter

Reply:
I'd go for

> Make all the Handle methods be explicit interface implementations, so I can keep the overload

😄